apps/web/components/setup/ChooseLicense.tsx

```diff
@@ -0,0 +1,78 @@
+import * as RadioGroup from "@radix-ui/react-radio-group";
+import classNames from "classnames";
+import { useRouter } from "next/router";
+import { Dispatch, SetStateAction } from "react";
+
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+
+const ChooseLicense = (props: {
+  currentStep: number;
+  isFreeLicense: boolean;
+  setIsLoading: Dispatch<SetStateAction<boolean>>;
+  setIsFreeLicense: Dispatch<SetStateAction<boolean>>;
+}) => {
```

Comment:
This components is caring for way more that it needs to:

```suggestion
const ChooseLicense = (
  props: {
    value: string;
    onChange: (value: string) => void;
    onSubmit: (value: EventTarget) => void;
  } & JSX.IntrinsicElements["form"]
) => {
```

Reply:
Actually, we need the `currentStep` to be able to let the form know which step is the submit button expecting to submit.

Reply:
Refactored to reduce prop drilling.

Reply:
> Actually, we need the `currentStep` to be able to let the form know which step is the submit button expecting to submit.

We only need ´currenStep´ to form the new href. This can happen on the parent hence the use of onSubmit instead.